Bottom: 98b0661f7b74d4b77513fb0b1e12b4fcf235c725
Top:    1eaf1e30fae8ce7753cb67b8dbac211565b6ca19
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-07 10:16:10 +0000

Attempt some GL optimisation


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index bf018ea..7617ce2 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -248,7 +248,7 @@ static void draw_cap (double width, int x, int y, double angle, double scale)
 void
 hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double scale)
 {
-  double angle;
+  double angle = 0.;
   float deltax, deltay, length;
   float wdx, wdy;
   int circular_caps = 0;
@@ -278,10 +278,15 @@ hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double s
     wdy = deltax * width / 2. / length;
     wdx = -deltay * width / 2. / length;
 
-    if (deltay == 0.)
-      angle = (deltax < 0) ? 270. : 90.;
+    if (hairline)
+      angle = 0;
     else
-      angle = 180. / M_PI * atanl (deltax / deltay);
+      {
+        if (deltay == 0.)
+          angle = (deltax < 0) ? 270. : 90.;
+        else
+          angle = 180. / M_PI * atanl (deltax / deltay);
+      }
 
     if (deltay < 0)
       angle += 180.;
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index a78c5fb..c232c07 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -905,6 +905,8 @@ ghid_drawing_area_expose_cb (GtkWidget *widget,
   glBlendFunc (GL_ONE, GL_ZERO);
 //  glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
+  glShadeModel (GL_FLAT);
+
   glViewport (0, 0, widget->allocation.width, widget->allocation.height);
 
   glEnable (GL_SCISSOR_TEST);
